### ISSUE TITLE
warning: [R.swift] Skipping 7 strings in 'Localizable' (Base) because symbol '_hours' would be generated for all of these keys: 2_hours, 3_hours, 4_hours, 5_hours, 6_hours, 7_hours, 8_hours

### ISSUE THREAD

**Issue by c0diq** [mac-cain13/R.swift#626]
My strings file has the following keys but are not being added to R.strings
```
"2_hours" = "2 hours";
"3_hours" = "3 hours";
"4_hours" = "4 hours";
"4_hours" = "4 hours";\
"5_hours" = "5 hours";\
"6_hours" = "6 hours";\
"7_hours" = "7 hours";
"8_hours" = "8 hours";
```
I'm getting the following errors:
```
warning: [R.swift] Skipping 7 strings in 'Localizable' (Base) because symbol '_hours' would be generated for all of these keys: 2_hours, 3_hours, 4_hours, 5_hours, 6_hours, 7_hours, 8_hours
```

It seems to appear for all keys that start with a number.

**Comment by c0diq** [comment]
The problem seems to be in `SwiftIdentifier.swift` caused by the line:
```
 // Remove numbers at the start of the name
    let sanitizedSwiftName = numberPrefixRegex.stringByReplacingMatches(in: cleanedSwiftName, options: [], range: cleanedSwiftName.fullRange, withTemplate: "")
```

Why is it doing that?

**Comment by mac-cain13** [comment]
Thanks for using R.swift!

I see how prefixing would help in your specific case, but this solution is not possible in all cases and would give inconsistent behavior.

Example:
`2_hours` & `3_hours` now both become `_hours` which you propose to fix by making it `_2_hours` that works just fine.
But `c#_note` & `c_note` now both become `c_note`, but that can't be fixed by a prefix.

I think changing the behavior makes it harder to understand. Therefore I prefer to keep it as is.

**Comment by c0diq** [comment]
I actually changed it to spell the number in the PR I submitted instead of prefixing it.

**Comment by mac-cain13** [comment]
It still doesn't give a consistent experience where R.swift converts strings like 0.5_inch or #5 into a logical easy to understand variable name. I believe a consistent easy to understand conversion to a variable name is more important and in most cases it's easy to change the name of your image/string/etc yourself.

**Comment by c0diq** [comment]
In some situations it is impractical or not possible to change keys across an entire codebase. We may not have access to some legacy source code where the ObjC accesses the keys directly. An option would be to duplicate keys in this case and making our string files bigger. 

The sanitization process already strips invalid characters from keys and converting to a variable name that does not match the string key necessarily. I agree that the sanitized key may not be as good as if it was changed in the strings file to not start with a digit but isn’t it better than the alternative to strip it entirely? I only saw this behavior because we had multiple keys starting with digits that once sanitized ended up causing collisions.
